local/local.go

```diff
@@ -22,6 +23,7 @@ type Image struct {
 	store          *Store
 	lastIdentifier string
 	daemonOS       string
+	mutex          sync.Mutex
```

Comment:
Why do we need the mutex? What's the race condition here?

Reply:
We are accessing [index manifest](https://github.com/buildpacks/imgutil/blob/6eb1a7027489a6508012fd461b27a5bd065e3780/local/store.go#L364) pointer concurrently, anyhow we are removing the MutateManifest due to image saving in legacy docker format, this is no more needed.

Reply:
Hmm... I'm still confused. When we're adding an image to a manifest list it's the index that's being mutated, not the image. So shouldn't the mutex live on the index?

Reply:
I believe we no longer need this, we can completely remove `MutateManifest` calls, which caused race condition here :(